netdisco/gdm.py

```diff
@@ -0,0 +1,98 @@
+"""
+Adds support for discovery using GDM (Good Day Mate),
+an in-house developped multicast protocol by Plex.
+
+Inspired by
+  hippojay's plexGDM:
+  https://github.com/hippojay/script.plexbmc.helper/resources/lib/plexgdm.py
+  iBaa's PlexConnect: https://github.com/iBaa/PlexConnect/PlexAPI.py
+"""
+import threading
+import socket
+
+
+class GDM(object):
+    """ Base class to discover GDM services. """
+
+    def __init__(self):
+        self.entries = []
+        self.last_scan = None
+        self._lock = threading.RLock()
+
+    def scan(self):
+        ''' Scan the network. '''
+        with self._lock:
+            self.update()
+
+    def all(self):
+        """
+        Returns all found entries.
+        Will scan for entries if not scanned recently.
+        """
+        self.scan()
+        return list(self.entries)
+
+    def find_by_content_type(self, value):
+        '''
+        Return a list of entries that match the content_type
+        '''
+        self.scan()
+        return [entry for entry in self.entries
+                if value in entry['data']['Content_Type']]
+
+    def find_by_data(self, values):
+        '''
+        Return a list of entries that match the search parameters
+        '''
+        self.scan()
+        return [entry for entry in self.entries
+                if (item in values.items() for item in entry['data'].items())]
```

Comment:
It should be the other way around. Instead of making sure all entries in `entry['data']` occurs in `values`, you want to have all entries in `values` to occur in `entry['data']`

```python
       return [entry for entry in self.entries
               if all(item in entry['data'].items() for item in values.items())]
```